docs/standards/v1/STANDARDS.md

```diff
@@ -0,0 +1,322 @@
+
+### Client Errors
+
+|Status Code|Description|Verbs|
+|---|---|---|
+|400 Bad Request|This status **MUST** be returned for requests that provide malformed or invalid data. Examples: malformed request body, unexpected query parameters, or invalid request fields.|GET, PATCH, POST, DELETE|
+|403 Forbidden|This status **MUST** be returned if the request cannot be performed by the user due to lack of permissions. Example: User with read-only permissions to a resource tries to update it. |POST, PATCH, DELETE|
+|404 Not Found|This status **MUST** be returned if the requested resource does not exist or if the user requesting the resource has insufficient permissions to view the resource.|GET, POST, PATCH, DELETE|
+
+### Server Errors
+
+|Status Code|Description|
+|---|---|
+|500 Internal Server Error|This status **MUST** be returned when a server error occurs.
+
+## Resources
+
+A resource represents an individual object within the system.  It is represented as a JSON object.  
+
+## Field Names
+
+Resource Fields will be camelCase.
+
+Resource Fields MUST include ONLY the following characters:
+
+* `a-z` (lowercase alpha)
+* `A-Z` (uppercase alpha)
+
+Resource fields that accept multiple values MUST be pluralized.
+
+#### Recommended Technologies to Support
+
+* https://github.com/vbabiy/djangorestframework-camel-case
+
+## Collections
+
+A collection is a list of multiple Resources. A collection is represented as a JSON object.
+
+If a collection is larger then per_page Pagination size then it `MUST` contain a links object, containing pagination information.
+
+
+## Pagination
+
+Pagination should be used by Collections to limit the number of resources returned at a time.
+
+Pagination is requested by a client through the use of query parameters.
+
+Requests will accept the following query parameters:
+
+- limit: maximum number of items to return
+- offset: starting position of the query in relation to the complete set of unpaginated items
+
+A paginated response would look as follows:
+
+```json
+{
+    "count": 1023,
+    "next": "https://api.example.org/accounts/?limit=100&offset=500",
+    "previous": "https://api.example.org/accounts/?limit=100&offset=300",
+    "results": [
+       …
+    ]
+}
+```
+#### Recommended Technologies to Support
+
+To achieve this, register a `DEFAULT_PAGINATION_CLASS` in your `local_settings/settings.py` to apply to all apps:
+
+```python
+REST_FRAMEWORK = {
+    "DEFAULT_PAGINATION_CLASS": "rest_framework.pagination.LimitOffsetPagination",
+    "PAGE_SIZE": 10
+}
+```
+
+PAGE_SIZE should be set to:  "TO BE DETERMINED"
+
+## Query Parameters
+
+Query Parameters **MUST** include **ONLY** the following characters:
+
+* `a-z` (lowercase only)
+* `_` (underscore)
+
+Query parameters that accept multiple values **MUST** be pluralized.
+
+If any request receives a query parameter it does not understand, the response **MUST** be a `400 Bad Request`.
```

Comment:
This is an interesting one. 

I see the benefit of being strict here, but I can see it also being a pain from an API consumer standpoint. Sometimes it's nice to serialize an entire object as query params without having to cherry-pick out any values that aren't valid.

This is how GitHub's API works: https://api.github.com/orgs/hackoregon/repos?sort=updated&beepboop=boopbeep!

It's also inline with the [robustness principle](https://en.wikipedia.org/wiki/Robustness_principle).

Meanwhile the [JSON:API spec](https://jsonapi.org/format/#query-parameters) doesn't take a stance on query parameters as long as there is no chance they could conflict with params that may be added to a future spec.

The [OpenAPI spec](http://spec.openapis.org/oas/v3.0.2) also doesn't seem concerned about extraneous query params.

Reply:
Removed requirement:

https://github.com/hackoregon/2019-backend-cookiecutter-django/commit/6d48f23f5e2422f87decaafd4475954f845ebb70